File: scenarios/basic/mobile/android-views/StatusByShapeOnly.java
package com.example.accessibilityscenarios;

import android.app.Activity;
import android.graphics.drawable.GradientDrawable;
import android.os.Bundle;
import android.view.View;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.ScrollView;
import android.widget.TextView;

public class StatusByShapeOnlyActivity extends Activity {
    
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        
        // Create main layout
        ScrollView scrollView = new ScrollView(this);
        LinearLayout mainLayout = new LinearLayout(this);
        mainLayout.setOrientation(LinearLayout.VERTICAL);
        mainLayout.setPadding(32, 32, 32, 32);
        mainLayout.setBackgroundColor(0xFFF5F5F5);
        
        // Header
        LinearLayout headerLayout = createHeader();
        mainLayout.addView(headerLayout);
        
        // Legend - Missing text descriptions
        LinearLayout legendLayout = createLegend();
        mainLayout.addView(legendLayout);
        
        // Documents List
        LinearLayout documentsLayout = createDocumentsList();
        mainLayout.addView(documentsLayout);
        
        scrollView.addView(mainLayout);
        setContentView(scrollView);
    }
    
    private LinearLayout createHeader() {
        LinearLayout headerLayout = new LinearLayout(this);
        headerLayout.setOrientation(LinearLayout.VERTICAL);
        headerLayout.setBackgroundColor(0xFF2196F3);
        headerLayout.setPadding(32, 32, 32, 32);
        
        TextView titleText = new TextView(this);
        titleText.setText("Document Management");
        titleText.setTextSize(24);
        titleText.setTextColor(0xFFFFFFFF);
        titleText.setTypeface(null, android.graphics.Typeface.BOLD);
        
        TextView subtitleText = new TextView(this);
        subtitleText.setText("Track document approval status");
        subtitleText.setTextSize(16);
        subtitleText.setTextColor(0xE6FFFFFF);
        
        headerLayout.addView(titleText);
        headerLayout.addView(subtitleText);
        
        return headerLayout;
    }
    
    private LinearLayout createLegend() {
        LinearLayout legendLayout = new LinearLayout(this);
        legendLayout.setOrientation(LinearLayout.VERTICAL);
        legendLayout.setBackgroundColor(0xFFFFFFFF);
        legendLayout.setPadding(16, 16, 16, 16);
        
        TextView legendTitle = new TextView(this);
        legendTitle.setText("Status Legend");
        legendTitle.setTextSize(16);
        legendTitle.setTypeface(null, android.graphics.Typeface.BOLD);
        legendTitle.setPadding(0, 0, 0, 12);
        
        legendLayout.addView(legendTitle);
        
        LinearLayout itemsLayout = new LinearLayout(this);
        itemsLayout.setOrientation(LinearLayout.HORIZONTAL);
        
        // STATUS BY SHAPE ONLY - No text descriptions
        LinearLayout approvedLayout = createLegendItem("circle", 0xFF4CAF50, "✓", "Approved");
        LinearLayout pendingLayout = createLegendItem("square", 0xFFFF9800, "⏰", "Pending");
        LinearLayout rejectedLayout = createLegendItem("square", 0xFFE53E3E, "✕", "Rejected");
        
        LinearLayout.LayoutParams params = new LinearLayout.LayoutParams(0, 
            LinearLayout.LayoutParams.WRAP_CONTENT, 1.0f);
        itemsLayout.addView(approvedLayout, params);
        itemsLayout.addView(pendingLayout, params);
        itemsLayout.addView(rejectedLayout, params);
        
        legendLayout.addView(itemsLayout);
        
        return legendLayout;
    }
    
    private LinearLayout createLegendItem(String shape, int color, String icon, String text) {
        LinearLayout itemLayout = new LinearLayout(this);
        itemLayout.setOrientation(LinearLayout.VERTICAL);
        itemLayout.setGravity(android.view.Gravity.CENTER_HORIZONTAL);
        
        // Status indicator by shape only
        LinearLayout statusShape = new LinearLayout(this);
        statusShape.setMinimumWidth(48);
        statusShape.setMinimumHeight(48);
        statusShape.setGravity(android.view.Gravity.CENTER);
        
        GradientDrawable shapeDrawable = new GradientDrawable();
        shapeDrawable.setColor(color);
        if (shape.equals("circle")) {
            shapeDrawable.setCornerRadius(24); // Circle = Approved
        } else {
            shapeDrawable.setCornerRadius(8); // Square = Pending/Rejected
        }
        statusShape.setBackground(shapeDrawable);
        
        TextView iconText = new TextView(this);
        iconText.setText(icon);
        iconText.setTextColor(0xFFFFFFFF);
        iconText.setTextSize(16);
        statusShape.addView(iconText);
        
        TextView labelText = new TextView(this);
        labelText.setText(text); // MISSING: Should be programmatically associated
        labelText.setTextSize(12);
        labelText.setTextColor(0xFF666666);
        labelText.setPadding(0, 8, 0, 0);
        
        itemLayout.addView(statusShape);
        itemLayout.addView(labelText);
        
        return itemLayout;
    }
    
    private LinearLayout createDocumentsList() {
        LinearLayout documentsLayout = new LinearLayout(this);
        documentsLayout.setOrientation(LinearLayout.VERTICAL);
        documentsLayout.setBackgroundColor(0xFFFFFFFF);
        documentsLayout.setPadding(16, 16, 16, 16);
        
        String[][] documents = {
            {"Project Proposal.docx", "approved", "2024-01-15", "2.4 MB"},
            {"Budget Report.xlsx", "pending", "2024-01-14", "1.8 MB"},
            {"Meeting Notes.pdf", "rejected", "2024-01-13", "856 KB"},
            {"Contract Template.docx", "approved", "2024-01-12", "3.2 MB"},
            {"Invoice #12345.pdf", "pending", "2024-01-11", "1.1 MB"}
        };
        
        for (String[] doc : documents) {
            LinearLayout docItem = createDocumentCard(doc[0], doc[1], doc[2], doc[3]);
            documentsLayout.addView(docItem);
        }
        
        return documentsLayout;
    }
    
    private LinearLayout createDocumentCard(String name, String status, String date, String size) {
        LinearLayout cardLayout = new LinearLayout(this);
        cardLayout.setOrientation(LinearLayout.HORIZONTAL);
        cardLayout.setBackgroundColor(0xFFF8F9FA);
        cardLayout.setPadding(16, 16, 16, 16);
        
        // Document Icon
        TextView docIcon = new TextView(this);
        docIcon.setText("📄");
        docIcon.setTextSize(20);
        docIcon.setPadding(0, 0, 12, 0);
        
        // Document Info
        LinearLayout infoLayout = new LinearLayout(this);
        infoLayout.setOrientation(LinearLayout.VERTICAL);
        
        TextView nameText = new TextView(this);
        nameText.setText(name);
        nameText.setTextSize(16);
        nameText.setTypeface(null, android.graphics.Typeface.BOLD);
        
        TextView detailsText = new TextView(this);
        detailsText.setText(date + " • " + size);
        detailsText.setTextSize(12);
        detailsText.setTextColor(0xFF666666);
        
        infoLayout.addView(nameText);
        infoLayout.addView(detailsText);
        
        // Status Indicator - STATUS BY SHAPE ONLY
        LinearLayout statusShape = new LinearLayout(this);
        statusShape.setMinimumWidth(48);
        statusShape.setMinimumHeight(48);
        statusShape.setGravity(android.view.Gravity.CENTER);
        
        GradientDrawable shapeDrawable = new GradientDrawable();
        TextView statusIcon = new TextView(this);
        statusIcon.setTextColor(0xFFFFFFFF);
        statusIcon.setTextSize(16);
        
        if (status.equals("approved")) {
            shapeDrawable.setColor(0xFF4CAF50);
            shapeDrawable.setCornerRadius(24); // Circle shape
            statusIcon.setText("✓");
        } else if (status.equals("pending")) {
            shapeDrawable.setColor(0xFFFF9800);
            shapeDrawable.setCornerRadius(8); // Square shape
            statusIcon.setText("⏰");
        } else {
            shapeDrawable.setColor(0xFFE53E3E);
            shapeDrawable.setCornerRadius(8); // Square shape
            statusIcon.setText("✕");
        }
        
        statusShape.setBackground(shapeDrawable);
        statusShape.addView(statusIcon);
        statusShape.setContentDescription(null); // MISSING: No accessible description for status
        
        LinearLayout.LayoutParams iconParams = new LinearLayout.LayoutParams(
            LinearLayout.LayoutParams.WRAP_CONTENT, 
            LinearLayout.LayoutParams.WRAP_CONTENT);
        cardLayout.addView(docIcon, iconParams);
        
        LinearLayout.LayoutParams infoParams = new LinearLayout.LayoutParams(0, 
            LinearLayout.LayoutParams.WRAP_CONTENT, 1.0f);
        cardLayout.addView(infoLayout, infoParams);
        
        LinearLayout.LayoutParams statusParams = new LinearLayout.LayoutParams(
            LinearLayout.LayoutParams.WRAP_CONTENT, 
            LinearLayout.LayoutParams.WRAP_CONTENT);
        cardLayout.addView(statusShape, statusParams);
        
        return cardLayout;
    }
}

File: scenarios/basic/mobile/android-compose/StatusByShapeOnly.kt
package com.example.accessibilityscenarios

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp

@Composable
fun StatusByShapeOnlyScreen() {
    val documents = listOf(
        Document(
            id = 1,
            title = "Project Proposal.docx",
            status = DocumentStatus.APPROVED,
            lastModified = "2024-01-15",
            size = "2.4 MB"
        ),
        Document(
            id = 2,
            title = "Budget Report.xlsx",
            status = DocumentStatus.PENDING,
            lastModified = "2024-01-14",
            size = "1.8 MB"
        ),
        Document(
            id = 3,
            title = "Meeting Notes.pdf",
            status = DocumentStatus.REJECTED,
            lastModified = "2024-01-13",
            size = "856 KB"
        ),
        Document(
            id = 4,
            title = "Contract Template.docx",
            status = DocumentStatus.APPROVED,
            lastModified = "2024-01-12",
            size = "3.2 MB"
        ),
        Document(
            id = 5,
            title = "Invoice #12345.pdf",
            status = DocumentStatus.PENDING,
            lastModified = "2024-01-11",
            size = "1.1 MB"
        )
    )
    
    Column(
        modifier = Modifier
            .fillMaxSize()
            .background(Color(0xFFF5F5F5))
            .padding(16.dp)
    ) {
        // Header
        Card(
            modifier = Modifier
                .fillMaxWidth()
                .padding(bottom = 16.dp),
            colors = CardDefaults.cardColors(containerColor = Color(0xFF2196F3))
        ) {
            Column(
                modifier = Modifier.padding(20.dp)
            ) {
                Text(
                    text = "Document Management",
                    fontSize = 24.sp,
                    fontWeight = FontWeight.Bold,
                    color = Color.White
                )
                Text(
                    text = "Track document approval status",
                    fontSize = 16.sp,
                    color = Color.White.copy(alpha = 0.9f)
                )
            }
        }
        
        // Legend - Missing text descriptions
        Card(
            modifier = Modifier
                .fillMaxWidth()
                .padding(bottom = 16.dp)
        ) {
            Column(
                modifier = Modifier.padding(16.dp)
            ) {
                Text(
                    text = "Status Legend",
                    fontSize = 16.sp,
                    fontWeight = FontWeight.Bold,
                    modifier = Modifier.padding(bottom = 12.dp)
                )
                
                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.SpaceEvenly
                ) {
                    // STATUS BY SHAPE ONLY - No text descriptions
                    Column(
                        horizontalAlignment = Alignment.CenterHorizontally
                    ) {
                        Box(
                            modifier = Modifier
                                .size(24.dp)
                                .background(Color(0xFF4CAF50), RoundedCornerShape(12.dp)), // Circle = Approved
                            contentAlignment = Alignment.Center
                        ) {
                            Icon(
                                imageVector = Icons.Default.Check,
                                contentDescription = null, // MISSING: No accessible description
                                tint = Color.White,
                                modifier = Modifier.size(16.dp)
                            )
                        }
                        Text(
                            text = "Approved", // MISSING: Should be programmatically associated
                            fontSize = 12.sp,
                            color = Color.Gray
                        )
                    }
                    
                    Column(
                        horizontalAlignment = Alignment.CenterHorizontally
                    ) {
                        Box(
                            modifier = Modifier
                                .size(24.dp)
                                .background(Color(0xFFFF9800), RoundedCornerShape(4.dp)), // Square = Pending
                            contentAlignment = Alignment.Center
                        ) {
                            Icon(
                                imageVector = Icons.Default.Schedule,
                                contentDescription = null, // MISSING: No accessible description
                                tint = Color.White,
                                modifier = Modifier.size(16.dp)
                            )
                        }
                        Text(
                            text = "Pending", // MISSING: Should be programmatically associated
                            fontSize = 12.sp,
                            color = Color.Gray
                        )
                    }
                    
                    Column(
                        horizontalAlignment = Alignment.CenterHorizontally
                    ) {
                        Box(
                            modifier = Modifier
                                .size(24.dp)
                                .background(Color(0xFFE53E3E), RoundedCornerShape(4.dp)), // Square = Rejected
                            contentAlignment = Alignment.Center
                        ) {
                            Icon(
                                imageVector = Icons.Default.Close,
                                contentDescription = null, // MISSING: No accessible description
                                tint = Color.White,
                                modifier = Modifier.size(16.dp)
                            )
                        }
                        Text(
                            text = "Rejected", // MISSING: Should be programmatically associated
                            fontSize = 12.sp,
                            color = Color.Gray
                        )
                    }
                }
            }
        }
        
        // Documents List
        LazyColumn(
            verticalArrangement = Arrangement.spacedBy(8.dp)
        ) {
            items(documents) { document ->
                DocumentCard(document = document)
            }
        }
    }
}

@Composable
fun DocumentCard(document: Document) {
    Card(
        modifier = Modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(8.dp)),
        colors = CardDefaults.cardColors(containerColor = Color.White),
        elevation = CardDefaults.cardElevation(defaultElevation = 2.dp)
    ) {
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .padding(16.dp),
            verticalAlignment = Alignment.CenterVertically
        ) {
            // Document Icon
            Icon(
                imageVector = Icons.Default.Description,
                contentDescription = null,
                tint = Color(0xFF1976D2),
                modifier = Modifier.size(24.dp)
            )
            
            Spacer(modifier = Modifier.width(12.dp))
            
            // Document Info
            Column(
                modifier = Modifier.weight(1f)
            ) {
                Text(
                    text = document.title,
                    fontSize = 16.sp,
                    fontWeight = FontWeight.Medium
                )
                Text(
                    text = "${document.lastModified} • ${document.size}",
                    fontSize = 12.sp,
                    color = Color.Gray
                )
            }
            
            // Status Indicator - STATUS BY SHAPE ONLY
            when (document.status) {
                DocumentStatus.APPROVED -> {
                    Box(
                        modifier = Modifier
                            .size(24.dp)
                            .background(Color(0xFF4CAF50), RoundedCornerShape(12.dp)), // Circle shape
                        contentAlignment = Alignment.Center
                    ) {
                        Icon(
                            imageVector = Icons.Default.Check,
                            contentDescription = null, // MISSING: No accessible description for status
                            tint = Color.White,
                            modifier = Modifier.size(16.dp)
                        )
                    }
                }
                DocumentStatus.PENDING -> {
                    Box(
                        modifier = Modifier
                            .size(24.dp)
                            .background(Color(0xFFFF9800), RoundedCornerShape(4.dp)), // Square shape
                        contentAlignment = Alignment.Center
                    ) {
                        Icon(
                            imageVector = Icons.Default.Schedule,
                            contentDescription = null, // MISSING: No accessible description for status
                            tint = Color.White,
                            modifier = Modifier.size(16.dp)
                        )
                    }
                }
                DocumentStatus.REJECTED -> {
                    Box(
                        modifier = Modifier
                            .size(24.dp)
                            .background(Color(0xFFE53E3E), RoundedCornerShape(4.dp)), // Square shape
                        contentAlignment = Alignment.Center
                    ) {
                        Icon(
                            imageVector = Icons.Default.Close,
                            contentDescription = null, // MISSING: No accessible description for status
                            tint = Color.White,
                            modifier = Modifier.size(16.dp)
                        )
                    }
                }
            }
        }
    }
}

enum class DocumentStatus {
    APPROVED,
    PENDING,
    REJECTED
}

data class Document(
    val id: Int,
    val title: String,
    val status: DocumentStatus,
    val lastModified: String,
    val size: String
)
